src/app/pages/movie-details-page/movie-details-page.component.ts

```diff
@@ -54,17 +59,42 @@ export class MovieDetailsPageComponent implements OnInit {
           this.loading.movie = false;
         }, 100)
       }
+     
+    )
+     // @ts-ignore
+    await this.movieService.fetchMovieWatchedStatus(+this.route.snapshot.paramMap.get('id')).subscribe( 
+      (resp) => {
+        resp.body.filter((movie:any) => {
+         return movie.id == this.route.snapshot.paramMap.get('id')
+        });
+        setTimeout(()=> {
+          this.userMovie.viewInfo.checked=resp.body.filter((movie:any) => movie.id == this.route.snapshot.paramMap.get('id')).length?'checked':'';
```

Comment:
Could you please indend this 2 lignes, for someone that reads your code for the first time, it is really hard to understand it.

Reply:
The api need to return a boolean that indicate if the current movie is already been watched or not.